src/Mod/Draft/draftobjects/dimension.py

```diff
@@ -0,0 +1,373 @@
+# *                                                                         *
+# ***************************************************************************
+
+"""This module provides the object code for Draft DimensionStyle.
+"""
+## @package style_dimension
+# \ingroup DRAFT
+# \brief This module provides the object code for Draft DimensionStyle.
+
+import FreeCAD as App
+import math
+from PySide.QtCore import QT_TRANSLATE_NOOP
+import DraftGeomUtils
+import draftutils.gui_utils as gui_utils
+import draftutils.utils as utils
+from draftobjects.draft_annotation import DraftAnnotation
+from draftviewproviders.view_dimension import ViewProviderDimensionBase
+from draftviewproviders.view_dimension import ViewProviderLinearDimension
+from draftviewproviders.view_dimension import ViewProviderAngularDimension
+
+def make_dimension(p1,p2,p3=None,p4=None):
+    """makeDimension(p1,p2,[p3]) or makeDimension(object,i1,i2,p3)
+    or makeDimension(objlist,indices,p3): Creates a Dimension object with
+    the dimension line passign through p3.The current line width and color
+    will be used. There are multiple  ways to create a dimension, depending on
+    the arguments you pass to it:
+    - (p1,p2,p3): creates a standard dimension from p1 to p2
+    - (object,i1,i2,p3): creates a linked dimension to the given object,
+    measuring the distance between its vertices indexed i1 and i2
+    - (object,i1,mode,p3): creates a linked dimension
+    to the given object, i1 is the index of the (curved) edge to measure,
+    and mode is either "radius" or "diameter".
+    """
+    if not App.ActiveDocument:
+        App.Console.PrintError("No active document. Aborting\n")
+        return
+    obj = App.ActiveDocument.addObject("App::FeaturePython","Dimension")
+    LinearDimension(obj)
+    if App.GuiUp:
+        ViewProviderLinearDimension(obj.ViewObject)
+    if isinstance(p1,App.Vector) and isinstance(p2,App.Vector):
+        obj.Start = p1
+        obj.End = p2
+        if not p3:
+            p3 = p2.sub(p1)
+            p3.multiply(0.5)
+            p3 = p1.add(p3)
+    elif isinstance(p2,int) and isinstance(p3,int):
+        l = []
+        idx = (p2,p3)
+        l.append((p1,"Vertex"+str(p2+1)))
+        l.append((p1,"Vertex"+str(p3+1)))
+        obj.LinkedGeometry = l
+        obj.Support = p1
+        p3 = p4
+        if not p3:
+            v1 = obj.Base.Shape.Vertexes[idx[0]].Point
+            v2 = obj.Base.Shape.Vertexes[idx[1]].Point
+            p3 = v2.sub(v1)
+            p3.multiply(0.5)
+            p3 = v1.add(p3)
+    elif isinstance(p3,str):
+        l = []
+        l.append((p1,"Edge"+str(p2+1)))
+        if p3 == "radius":
+            #l.append((p1,"Center"))
+            if App.GuiUp:
+                obj.ViewObject.Override = "R $dim"
+            obj.Diameter = False
+        elif p3 == "diameter":
+            #l.append((p1,"Diameter"))
+            if App.GuiUp:
+                obj.ViewObject.Override = "Ø $dim"
```

Comment:
The Travis tests are failing in this line because of the unicode character `"Ø $dim"` . You have to add the codification at the beginning of the file. `# -*- coding: utf8 -*-`

Reply:
thanks for looking into this. I'll add it right now.

Reply:
shouldn't I add it to every file?

Reply:
I wouldn't add it to every file, only to the one reported by our tests. This is actually only required in Python 2. But if it causes this error in Travis, it may mask other errors elsewhere, so we try to have no errors at all when making a pull request.